### PR TITLE
[12.0] attachment_s3: allow to override connection parameters

### PR DESCRIPTION
**Pull request by grindtildeath**
_get_s3_connection_params can now be overrided to use different connection parameters

### CODE REVIEW

**Review by simahawk — APPROVED** [review]
LG

File: attachment_s3/models/ir_attachment.py [hunk 1]
```diff
@@ -78,6 +66,25 @@ def _get_s3_bucket(self, name=None):
                     ) % (bucket_name, bucket_name)
 
             raise exceptions.UserError(msg)
```
Comment (guewen) [inline attachment_s3/models/ir_attachment.py]: Shouldn't we raise this exception in _get_s3_bucket after we call `_get_s3_connection_params()`?
It allows a chance to call `super()` and adjust the parameters afterwards without getting an exception.

Comment (simahawk) [reply attachment_s3/models/ir_attachment.py]: hmm you make a good point. I agree.

**Review by simahawk — CHANGES_REQUESTED** [review]
@grindtildeath pls check Guewen's comment

**Comment by grindtildeath** [comment]
@simahawk Sorry for the delay :sweat_smile: 

Fixed and rebased, if you want to take a look :wink:

**Review by simahawk — APPROVED** [review]

**Pull request merged**